src/sidebar/util/page-range.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * Return true if the page number `label` is within `range`.
+ *
+ * @param label - A page number such as "10", "iv"
+ * @param range - A page range expressed as a single page number, or a hyphen
+ *   separated range (eg. "10-12"). Page ranges are inclusive, so the page
+ *   range "10-12" matches "10", "11" and "12". This means there is no way to
+ *   specify an empty range.
+ */
+export function pageLabelInRange(label: string, range: string): boolean {
+  if (range.includes('-')) {
+    let [start, end] = range.split('-');
+    if (!start) {
+      start = label;
+    }
+    if (!end) {
+      end = label;
+    }
+    const [startInt, endInt, labelInt] = [
+      parseInt(start),
+      parseInt(end),
+      parseInt(label),
+    ];
+    if (
+      Number.isInteger(startInt) &&
+      Number.isInteger(endInt) &&
+      Number.isInteger(labelInt)
+    ) {
+      return labelInt >= startInt && labelInt <= endInt;
+    } else {
+      return false;
+    }
+  } else {
+    return label === range;
+  }
```

Comment:
Perhaps we can simplify the conditions that need to be mentally dragged by adding an early return for the simplest condition, and adding the rest afterwards:

```suggestion
  if (!range.includes('-')) {
    return label === range;
  }
  
  let [start, end] = range.split('-');
  if (!start) {
    start = label;
  }
  if (!end) {
    end = label;
  }
  const [startInt, endInt, labelInt] = [
    parseInt(start),
    parseInt(end),
    parseInt(label),
  ];
  if (
    Number.isInteger(startInt) &&
    Number.isInteger(endInt) &&
    Number.isInteger(labelInt)
  ) {
    return labelInt >= startInt && labelInt <= endInt;
  }
  
  return false;
```

Reply:
I agree. I prefer the early-return version.